js/script.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Declarations
+ */
+const inputTitle = document.getElementById('inputTitle');
+const inputAuthor = document.getElementById('inputAuthor');
+const divBooks = document.querySelector('.books');
+const form = document.getElementsByTagName('form')[0];
+let booksCollection = [];
+const bookTemplate = `<div class="bookRow"><label id="title">{book.title}</label><label id="author">{book.author}</label>
+    <button type="button" class="remove" id="{book.id}" onclick="removeBook(this.id);">Remove</button>
+    </div><hr>`;
+
+/**
+     * Functions
+     */
+function Book(id, author, title) {
+  this.id = id;
+  this.title = title;
+  this.author = author;
+}
+
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+      // everything except Firefox
+      e.code === 22
+            // Firefox
+            || e.code === 1014
+            // test name field too, because code might not be present
+            // everything except Firefox
+            || e.name === 'QuotaExceededError'
+            // Firefox
+            || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+            // acknowledge QuotaExceededError only if there's something already stored
+            && (storage && storage.length !== 0);
+  }
+}
+function AddBook() {
+  let id = 1;
+  let books = localStorage.getItem('books');
+  if (books != null) {
+    booksCollection = JSON.parse(books);
+    const ids = booksCollection.map((object) => object.id);
+    const maxId = Math.max(...ids);
+    id = maxId + 1;
+  }
+
+  const objectBook = new Book(id.toString(), inputAuthor.value, inputTitle.value);
+  booksCollection.push(objectBook);
+
+  books = JSON.stringify(booksCollection);
+  localStorage.setItem('books', books);
+}
+
+function removeBook(id) {
+  if (id > 0) {
+    let books = localStorage.getItem('books');
+    if (books != null) {
+      booksCollection = JSON.parse(books);
+    }
+    const booksCollectionResult = booksCollection.filter((bk) => bk.id !== parseInt(id, 10));
+    booksCollection.splice();
+    books = JSON.stringify(booksCollectionResult);
+    localStorage.setItem('books', books);
+    window.location.reload();
+  }
+}
```

Comment:
- [ ] The remove button is not removing the book by id. It would be great to achieve this functionality. 😉